src/main/kotlin/org/rust/lang/core/resolve2/ModCollector.kt

```diff
@@ -487,6 +488,11 @@ private fun RsItemsOwner.getOrBuildStub(): StubElement<out RsItemsOwner>? {
      */
     (this as? StubBasedPsiElement<*>)?.greenStub?.let { return it as StubElement<out RsItemsOwner> }
     if (this is RsBlock) return buildStub()
+    if (this is RsModItem) {
+        val containingFile = containingFile as? RsFile ?: return null
+        containingFile.calcStubTree()
+        return greenStub
```

Comment:
Since PSI is linked with stubs via a SoftReference, in theory `greenStub` can still be `null` after  `calcStubTree()` invocation (but I don't believe such thing can happen in the reality). But I think I know how to make it infallible:

```suggestion
        val stubIndex = PsiAnchor.calcStubIndex(this)
        if (stubIndex == -1) return null
        return containingFile.calcStubTree().plainList[stubIndex] as StubElement<out RsItemsOwner>
```